internal/kv/etcd/etcd_kv_test.go

```diff
@@ -909,6 +910,94 @@ func TestEtcdKV_Load(te *testing.T) {
 	})
 }
 
+func Test_WalkWithPagination(t *testing.T) {
+	etcdCli, err := etcd.GetEtcdClient(
+		Params.EtcdCfg.UseEmbedEtcd.GetAsBool(),
+		Params.EtcdCfg.EtcdUseSSL.GetAsBool(),
+		Params.EtcdCfg.Endpoints.GetAsStrings(),
+		Params.EtcdCfg.EtcdTLSCert.GetValue(),
+		Params.EtcdCfg.EtcdTLSKey.GetValue(),
+		Params.EtcdCfg.EtcdTLSCACert.GetValue(),
+		Params.EtcdCfg.EtcdTLSMinVersion.GetValue())
+	defer etcdCli.Close()
+	assert.NoError(t, err)
+
+	rootPath := "/etcd/test/root/pagination"
+	etcdKV := etcdkv.NewEtcdKV(etcdCli, rootPath)
+
+	defer etcdKV.Close()
+	defer etcdKV.RemoveWithPrefix("")
+
+	kvs := map[string]string{
+		"t/key_1": "value_1",
+		"t/key_2": "value_2",
+		"t/key_3": "value_3",
+		"t/key_4": "value_4",
+		"t/key_5": "value_5",
+	}
+
+	err = etcdKV.MultiSave(kvs)
+	assert.NoError(t, err)
+	for k, v := range kvs {
+		actualV, err := etcdKV.Load(k)
+		assert.NoError(t, err)
+		assert.Equal(t, v, actualV)
+	}
+
+	t.Run("apply function error ", func(t *testing.T) {
+		err = etcdKV.WalkWithPagination("t", 6, func(key []byte, value []byte) error {
+			return errors.New("error")
+		})
+		assert.Error(t, err)
+	})
+
+	t.Run("get with non-exist prefix ", func(t *testing.T) {
+		err = etcdKV.WalkWithPagination("non-exist-prefix", 6, func(key []byte, value []byte) error {
+			return nil
+		})
+		assert.NoError(t, err)
+	})
+
+	t.Run("no pagination ", func(t *testing.T) {
+		err = etcdKV.WalkWithPagination("t", 6, func(key []byte, value []byte) error {
+			k := string(key)
+			k = k[len(rootPath)+1:]
+			_, ok := kvs[k]
+			assert.True(t, ok)
+			return nil
+		})
+		assert.NoError(t, err)
+	})
+
+	t.Run("with pagination-2 ", func(t *testing.T) {
+		cnt := 0
```

Comment:
the test result is not correct, and the implementation is not correct.

Here we expect the result to be 6, not 2.

Reply:
pagination-2 just is a test name,  it will test for pagination is 2, and it will walk all data, it works for me
![image](https://user-images.githubusercontent.com/4024711/210221744-d26b8fe9-3ded-4dd0-81d7-21dd5bbc43d1.png)